### DISCUSSION TITLE
require('ts-pattern') not work on webpack5 but work on webpack4.

### DISCUSSION THREAD

**Discussion by ChengOuyang** [webpack/webpack#16250] [Q&A]
# Bug report

**What is the current behavior?**

const tsPattern = require('ts-pattern');

The tsPattern is a string but not a module on webpack5, and other library is work.

**If the current behavior is a bug, please provide the steps to reproduce.**

1. npx create-react-app ts-pattern-test
2. cd ts-pattern-test
3. pnpm add ts-pattern
4. code in src/index.js:
```
const tsPattern = require('ts-pattern');
```

**What is the expected behavior?**

**Other relevant information:**
webpack version:  react-scripts 5.0.1
Node.js version: 
Operating System: 
Additional tools:

**Comment by alexander-akait** [comment]
Please provide more infromation, an export is `export{S as P,S as Pattern,j as isMatching,K as match};`, so I don't undestand `The tsPattern is a string but not a module on webpack5, and other library is work.`

**Reply by ChengOuyang** [reply]
> 
```
const tsPattern = require('ts-pattern'); // tsPattern is "/static/media/index.dab05382817b14a76cb3.cjs"

// The following code will throw an error：
const { match } = tsPattern;
```

Reproduce the project below：
https://github.com/ChengOuyang/ts-pattern-test

**Comment by ChengOuyang** [comment]
This problem may not be a problem with webpack5, but may be cra：

https://github.com/facebook/create-react-app/issues/12725

**Reply by alexander-akait** [reply]
Yeah, this is a bug and this bug in CRA, my recommendation to use `import { Pattern } from "ts-pattern";`, because you don't have ability to change webpack configuration (because the bug is here).
